index.js

```diff
@@ -1,56 +1,147 @@
-const fastify = require('fastify')({ logger: true })
+const fastify = require("fastify")({ logger: true });
+const rateLimit = require('@fastify/rate-limit');
+const helmet = require('@fastify/helmet');
+const authPlugin = require("./authPlugin");
 
-const items = [{
-        id: 1,
-        name: 'Cartable',
-        price: 4
-}]
+const items = [
+  {
+    id: 1,
+    name: "Cartable",
+    price: 4,
+  },
+];
+
+const users = [
+	{
+		mail: 'enzo',
+		password: 'MotdePasse#'
+	},
+	{
+		mail: 'rick',
+		password: 'MotdePasse#'
+	}
+];
+
+fastify.register(authPlugin, {
+	users: users
+});
+
+fastify.register(rateLimit, {
+	max: 7,
+	timeWindow: '30 secondes'
+});
+
+fastify.register(helmet,{ 
+	contentSecurityPolicy: false 
+});
 
 // Declare a route
-fastify.get('/', async (request, reply) => {
-        const qs = request.query;
-        if (qs.from === 'jb') {
-                return `Hello ${qs.from}: here's your secret!`;
-        } else {
-                reply.code(401).send('Unauthorized')
-        }
-})
-
-fastify.get('/products', async (request, reply) => {
-        return items;
-})
-
-fastify.get('/products/:id', async (request, reply) => {
-        const id = parseInt(request.params.id, 10);
-        const desiredItem = items.find((item) => item.id === id);
-        if (desiredItem) {
-                return desiredItem;
-        } else {
-                reply.code(404).send('not found');
-        }
-});
-
-fastify.put('/products/:id', async (request, reply) => {
-        const id = parseInt(request.params.id, 10);
-        if (!request.body.name || !request.body.price) {
-                reply.code(400).send('Product must have a name and a price')
-        } else {
-                items.push({
-                        id: id,
-                        name: request.body.name,
-                        price: request.body.price
-                });
-                return reply.code(201).send(`Product ${request.body.name} has been created`);
-        }
-})
+fastify.get("/", async (request, reply) => {
+  const qs = request.query;
+	return `hello ${request.user}`;
+});
+
+fastify.get("/products", async (request, reply) => {
+	return items;
+});
+
+fastify.get("/products/:id", async (request, reply) => {
+  const id = parseInt(request.params.id, 10);
+  const desiredItem = items.find((item) => item.id === id);
+  if (desiredItem) {
+    return desiredItem;
+  } else {
+    reply.code(404).send("not found");
+  }
+});
+
+fastify.put("/products/:id", async (request, reply) => {
+  const id = parseInt(request.params.id, 10)
+
+  if (!request.body.name || !request.body.price) {
+    reply.code(400).send("Product must have a name and a price");
+  } else {
+		const existingPorductKey = items.findIndex((item) => item.id === id);
+
+    if(existingPorductKey != -1){
+      items[existingPorductKey] = {
+				id: id,
+				name: request.body.name,
+				price: request.body.price,
+			}
+			return reply
+				.code(201)
```

Comment:
```suggestion
				.code(200)
```
`201` est réservé à la création